app/pages/lab/workflow-viewer/answers.cjsx

```diff
@@ -0,0 +1,58 @@
+React = require 'react'
+Sty = require './jsplumb-style'
+icons = require '../../../classifier/tasks/drawing/icons'
+{Markdown} = require 'markdownz'
+
+# Styling for ansers within a list
+# AnswerItem takes in `input`, `eps`, and `jp` as props
+# `jp` is the jsPlumb instance for the visualizer
+# `eps` = {add: add endput callback, remove: remove endpoint callback, zIndex: the z-index of the task, endpionts: list of all endpoints}
+# `inputs` = {listId: unique uuid for endpoint, item: answer or tool, type: type of task, idx: current index of answer or tool}
+AnswerItem = React.createClass
+  displayName: 'AnswerItem'
+
+  # check if a string is just whitespace
+  isEmptyStr: (str) ->
+    if str.replace(/^\s+|\s+$/g, '').length == 0
+      return '&nbsp;'
+    else
+      return str
+
+  render: ->
+    if @props.inputs.type == 'drawing'
+      icon = <span className='drawing-tool-icon-vis' style={color: @props.inputs.item.color}>{icons[@props.inputs.item.type]}</span>
+    else
+      icon = undefined
+    <li className='answer-item' id={@props.inputs.listId}>
+      <div className='lab'>
+        {icon}
+        <Markdown className='markdown-vis'>{@isEmptyStr(@props.inputs.item.label)}</Markdown>
+      </div>
+    </li>
+
+# Populate a list containing tools/answers for a task
+# props are `plumbId`, `inputs`, `task`, and `jp`
+# `plumbId` is a unique ID for the task
+# `jp` is the jsPlumb instance for the visualizer
+# `task` is the workflow task
+# `inputs` = {uuid: {set: set a new uuid, get: get a new/existing uuid}, eps: {add: add endput callback, remove: remove endpoint callback, zIndex: the z-index of the task, endpionts: list of all endpoints}}
+module.exports = React.createClass
+  displayName: 'AnswerList'
+
+  createAnswer: (idx, item) ->
+    inputs =
+      idx: idx
+      item: item
+      listId: @props.inputs.uuid.get(idx)
+      type: @props.task.type
+
+    <AnswerItem jp={@props.jp} key={"Al_#{idx}"} inputs={inputs} eps={@props.inputs.eps} />
+
+  render: ->
+    if @props.task.type == 'drawing'
+      answers = (@createAnswer(idx, item) for idx, item of @props.task.tools)
+    else
+      answers = (@createAnswer(idx, item) for idx, item of @props.task.answers)
```

Comment:
I think this section would be a bit easier to understand if you moved the conditional into `createAnswer`, especially cause it already reaches into props rather than spreading the logic across both render and createAnswer. Maybe adding `@props.task` as an argument to the function?